cli/src/commands/next.rs

```diff
@@ -115,24 +115,18 @@ pub(crate) fn cmd_next(
             .view()
             .heads()
             .contains(current_wc_id);
-    let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
-    // If we're editing, start at the working-copy commit.
-    // Otherwise start from our direct parent.
-    let start_id = if edit {
-        current_wc_id
+    let wc_revset = RevsetExpression::commit(current_wc_id.clone());
+    // If we're editing, start at the working-copy commit. Otherwise, start from
+    // its direct parent(s).
+    let target_revset = if edit {
+        wc_revset.descendants_at(args.offset)
     } else {
-        match current_wc.parent_ids() {
-            [parent_id] => parent_id,
-            _ => return Err(user_error("Cannot run `jj next` on a merge commit")),
-        }
-    };
-    let descendant_expression = RevsetExpression::commit(start_id.clone()).descendants_at(offset);
-    let target_expression = if edit {
-        descendant_expression
-    } else {
-        descendant_expression.minus(&RevsetExpression::commit(current_wc_id.clone()).descendants())
+        wc_revset
+            .parents()
+            .descendants_at(args.offset)
+            .minus(&wc_revset)
```

Comment:
Did you remove `wc_revset.descendants()` by intention?

I think `.descendants()` was redundant because `--edit` is implied now, but it's worth mentioning.

Reply:
Can you clarify the question? I'm not sure I understand what you're asking about.

If you're asking about the `.minus`, that's intentional.

Reply:
The query is changed from `.minus(&wc_revset.descendants())` to `.minus(&wc_revset)`.

The original query was more correct because `@::` shouldn't be considered the "next" candidates, but that wouldn't matter anymore since `--edit` is implied if `@` isn't a head. The new query is optimal.

While I think the new query is better, this is subtle change, and should be documented in commit description or comment.

Reply:
> but that wouldn't matter **anymore** since --edit is implied if @ isn't a head.

Ah. I wasn't around when this command was originally implemented and I thought it always worked this way. Looking through the commit history it looks like this is actually a pretty recent change from https://github.com/martinvonz/jj/pull/3028.

For what it's worth I'm not sure if the old behavior for `next` would be very ergonomic since you could be in the middle of a linear history and `jj next` would refuse to move.

I added a comment above the call to `.minus()`.